src/Data/Functor/Foldable.hs

```diff
@@ -843,6 +828,12 @@ instance GCoerce f g => GCoerce (M1 i c f) (M1 i c' g) where
 instance GCoerce (K1 i c) (K1 j c) where
     gcoerce = K1 . unK1
 
+instance GCoerce U1 U1 where
+    gcoerce = id
+
+instance GCoerce V1 V1 where
+    gcoerce = id
```

Comment:
While we're at it, we may as well support datatype with zero constructors, even though I don't think it's going to be very useful.